sonar-plugin-api/src/main/java/org/sonar/api/batch/bootstrap/ProjectBuilder.java

```diff
@@ -33,11 +33,13 @@
  *   <li>Change project metadata like description or source directories.</li>
  * </ul>
  *
+ * @deprecated since 6.5. It won't be possible to manipulate the project's structure.
  * @since 2.9
  */
 @ScannerSide
 @InstantiationStrategy(InstantiationStrategy.PER_BATCH)
 @ExtensionPoint
+@Deprecated
 public abstract class ProjectBuilder {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)